k9mail/src/main/java/com/fsck/k9/mailstore/LockableDatabase.java

```diff
@@ -380,6 +372,7 @@ private void openOrCreateDataspace() throws UnavailableStorageException {
             } catch (SQLiteException e) {
                 // try to gracefully handle DB corruption - see issue 2537
                 Log.w(K9.LOG_TAG, "Unable to open DB " + databaseFile + " - removing file and retrying", e);
+                // noinspection ResultOfMethodCallIgnored, not much we can do if this fails
```

Comment:
I disagree in this case, since for this way of handling this particular exception, the behavior correct - the file should be deleted if possible, but if that fails there is nothing that could be done about it (outside of logging, maybe).

If the exception handling itself should be changed, a TODO is the way to go.